numpy/core/code_generators/generate_umath.py

```diff
@@ -280,7 +287,7 @@ def english_upper(s):
     Ufunc(2, 1, Zero,
           docstrings.get('numpy.core.umath.add'),
           'PyUFunc_AdditionTypeResolver',
-          TD(notimes_or_obj, simd=[('avx512f', cmplxvec),('avx2', ints)]),
+          TD(notimes_or_obj, simd=[('avx2', ints)], dispatch=[('loops_arithm_fp', 'fdFD')]),
```

Comment:
Just to be clear: this adds `fd` (`cmplxvec` is `FD`). Also: I don't see where 'gG' (long double and long double complex) is taken into account. Am I missing something?

Reply:
Yes, it dispatches single/double for both real and complex the definitions of these loops moved from `loops.c.src` and `simd.inc` to the new dispatch-able source `loops_arithm_fp.dispatch.c.src`.
NOTE: divide operation for complex single and double precision still in `loops.c.src`, we will need to vectorize this operation via NPYV and move it to this file later.

> I don't see where 'gG' (long double and long double complex) is taken into account. Am I missing something?

`loops_arithm_fp.dispatch.c.src` only contains inner loops functions with SIMD code the rest operations
remains in `loops.c.src`.
I don't think we will support SIMD kernels for `80-bit` and `128-bit` precision, there's no hardware support for it but 
maybe we can emulate light operations such as addition and subtract

Reply:
Ahh, my bad, it is in `notimes_or_obj`

Reply:
> 128-bit precision

 AArch64 and Power9 have support for quad precision floats. The astronomy community would like to have them and I think we will see more support coming in modern architectures.

Reply:
AArch64 and Power9 have scalar support but not SIMD. Power9 users have to build NumPy with CFLAGS `-mcpu=power9` or `--cpu-baseline=vsx3` in order to get `__float128` works, we can dispatch float128 operation in runtime for them if it's necessary.